framework/src/materials/Material.C

```diff
@@ -182,12 +187,45 @@ Material::getOutputs()
 }
 
 void
+Material::subdomainSetup()
+{
+  if (_constant_option == 2)
+  {
+    _current_subdomain_id = _current_elem->subdomain_id();
+
+    unsigned int nqp = _fe_problem.getMaxQps();
+
+    MaterialProperties & props = _material_data->props();
+    for (const auto & prop_id : _supplied_prop_ids)
+      props[prop_id]->resize(nqp);
+
+    _qp = 0;
+    computeSubdomainProperties();
+
+    for (const auto & prop_id : _supplied_prop_ids)
+    {
+      for (decltype(nqp) qp = 1; qp < nqp; ++qp)
+        props[prop_id]->qpCopy(qp, props[prop_id], 0);
+    }
+  }
+}
+
+void
+Material::computeSubdomainProperties()
```

Comment:
@andrsd, @friedmud - Just getting back around to this whole constant subdomain material property issue. As you can see, @YaqiWang has taken a stab and this and it's much better than before. This is a pretty fundamental change though to the framework so I want to make sure you two buy off before we go any further.